.github/workflows/tests.yml

```
@@ -39,7 +25,36 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # detect whether any code changes are included in this PR
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      code: ${{ steps.filter.outputs.code }}
+    steps:
+      - uses: actions/checkout@v3
+        # dorny/paths-filter needs git clone for push events
+        # https://github.com/marketplace/actions/paths-changes-filter#supported-workflows
+        if: github.event_name == 'push'
+      - uses: dorny/paths-filter@v2
```

We should pin to a commit here or fork the action into conda/actions.
```suggestion
      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50
```

---

Currently, across both conda and conda-build we utilize a number of actions that we do not fork into conda/actions:

- actions-ecosystem/action-add-labels
- actions-ecosystem/action-remove-labels
- actions/add-to-project
- actions/cache
- actions/checkout
- actions/download-artifact
- actions/github-script
- actions/setup-python
- actions/stale
- actions/upload-artifact
- andstor/file-existence-action
- codecov/codecov-action
- conda-incubator/setup-miniconda
- conda/actions/canary-release
- conda/actions/check-cla
- conda/actions/read-yaml
- dessant/lock-threads
- docker/build-push-action
- docker/login-action
- docker/metadata-action
- docker/setup-buildx-action
- docker/setup-qemu-action
- dorny/paths-filter
- EndBug/label-sync
- re-actors/alls-green
- test-summary/action

Are we expected to fork all of these other actions as well? I would strongly resist any such effort.

That said, I can get on the commit hash bandwagon. Do we want to open a new issue to track updating our workflows to use commit hashes for their `uses`?